wpilib/src/i2c.rs

```diff
@@ -0,0 +1,197 @@
+use std::io;
+use wpilib_sys::*;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[repr(i32)]
+pub enum Port {
+    Onboard = HAL_I2CPort::HAL_I2C_kOnboard,
+    MXP = HAL_I2CPort::HAL_I2C_kMXP,
+}
+
+pub struct I2C {
+    port: Port,
+    device_address: i32,
+}
+
+impl I2C {
+    /// Constructs a new I2C
+    ///
+    /// `port` is the I2C port to which the device is connected, and `device_address` is the address of the device on the bus
+    pub fn new(port: Port, device_address: i32) -> HalResult<Self> {
```

Comment:
So I2C addresses are either 7 bits or 10 bits. I'm not sure whether the roboRIO does 10-bit addressing, but I guess if it (or a future platform) does the HAL would abstract it away. Make the input `device_address` a u16 I guess?

Reply:
All of the HAL functions expect `i32` for the device address field, and clippy doesn't like potentially lossy casts from u16 to i32